swarm/version/version.go

```diff
@@ -51,6 +55,7 @@ func ArchiveVersion(gitCommit string) string {
 	}
 	if len(gitCommit) >= 8 {
 		vsn += "-" + gitCommit[:8]
+		GitCommit = gitCommit[:8]
```

Comment:
This is redundant. We can just use this function, it also gives the current version, next to the commit, which is not bad.

Reply:
Not only this is redundant, but actually wrong - `GitCommit` is set during build time, we should not be modifying it.